plugins/scaffolder-backend-module-gitlab/src/actions/createGitlabProjectAccessTokenAction.ts

```diff
@@ -35,46 +37,103 @@ export const createGitlabProjectAccessTokenAction = (options: {
+        accessLevel: z
+          .number({
+            description:
+              'Access Level of the Token, 10 (Guest), 20 (Reporter), 30 (Developer), 40 (Maintainer), and 50 (Owner)',
+          })
+          .optional(),
+        scopes: z
+          .array(
+            z.enum(
+              [
+                'api',
+                'read_api',
+                'create_runner',
+                'read_registry',
+                'write_registry',
+                'read_repository',
+                'write_repository',
+              ],
+              {
+                description: 'Scopes for a project access token',
+              },
+            ),
+          )
+          .optional(),
+        expiresAt: z
+          .string({
+            description:
+              'Expiration date of the access token in ISO format (YYYY-MM-DD). If Empty, it will set to the maximum of 365 days.',
+          })
+          .optional(),
+        token: z
+          .string({
+            description: 'The token to use for authorization to GitLab',
+          })
+          .optional(),
+      }),
       output: z.object({
         access_token: z.string({ description: 'Access Token' }),
       }),
     },
     async handler(ctx) {
       ctx.logger.info(`Creating Token for Project "${ctx.input.projectId}"`);
-      const { projectId, name, accessLevel, scopes } = ctx.input;
+      const {
+        projectId,
+        name = 'tokenname',
+        accessLevel = 40,
+        scopes = ['read_repository'],
+        expiresAt,
+      } = ctx.input;
+
       const { token, integrationConfig } = getToken(ctx.input, integrations);
 
-      const response = await fetch(
-        `${integrationConfig.config.baseUrl}/api/v4/projects/${projectId}/access_tokens`,
+      if (!integrationConfig.config.token && token) {
+        throw new InputError(
+          `No token available for host ${integrationConfig.config.baseUrl}`,
+        );
+      }
+
+      let api;
+
+      if (!ctx.input.token) {
+        api = new Gitlab({
+          host: integrationConfig.config.baseUrl,
+          token: token,
+        });
+      } else {
+        api = new Gitlab({
+          host: integrationConfig.config.baseUrl,
+          oauthToken: token,
+        });
+      }
+
+      const mappedAccessLevel = Number(accessLevel.valueOf()) as AccessLevel;
+
+      const response = await api.ProjectAccessTokens.create(
+        projectId,
+        name,
+        scopes,
         {
-          method: 'POST', // *GET, POST, PUT, DELETE, etc.
-          headers: {
-            'PRIVATE-TOKEN': token,
-            'Content-Type': 'application/json',
-          },
-          body: JSON.stringify({
-            name: name,
-            scopes: scopes,
-            access_level: accessLevel,
-          }),
+          expiresAt:
+            expiresAt || DateTime.now().plus({ days: 365 }).toISODate()!,
+          accessLevel: mappedAccessLevel,
```

Comment:
Not a bad thing to have expiration, but is it required from the GitLab API? Wondering if this might lead to unintended consequences for people that haven't set it yet & don't intend to have a expiration date.

Reply:
Read the original issue again - I assume this is required.